server/routes/users/friends/friends.route.js

```diff
@@ -0,0 +1,117 @@
+const express = require('express');
+const passport = require('passport');
+const mongoose = require('mongoose');
+const router = express.Router({ mergeParams: true });
+const User = require('../../../models/user');
+
+// @route:  GET /users/:userId/friends
+// @desc:   Find and return array of friends of specified userId
+// @access: Public
+router.get('/', async (req, res) => {
+  const user = await User.findById(req.params.userId);
+  const friends = [];
+
+  for (const friendId of user.friendIds) {
+    friends.push(await User.findById(friendId));
```

Comment:
This gets all the friends asynchronous one by one, and makes a lot of database API calls

Instead, you should make one call to Users, but in fact, you should use:
 https://mongoosejs.com/docs/populate.html

You also don't want to be returning all the fields from the database, that is a security issue